lib/util.js

```diff
@@ -662,28 +701,31 @@ function formatValue(ctx, value, recurseTimes) {
       // Fast path for ArrayBuffer and SharedArrayBuffer.
       // Can't do the same for DataView because it has a non-primitive
       // .buffer property that we need to recurse for.
-      const prefix = getPrefix(constructor, tag);
+      let prefix = getPrefix(constructor, tag);
+      if (prefix === '') {
+        prefix = isArrayBuffer(value) ? 'ArrayBuffer ' : 'SharedArrayBuffer ';
+      }
       if (keyLength === 0)
         return prefix +
               `{ byteLength: ${formatNumber(ctx.stylize, value.byteLength)} }`;
       braces[0] = `${prefix}{`;
       keys.unshift('byteLength');
     } else if (isDataView(value)) {
-      braces[0] = `${getPrefix(constructor, tag)}{`;
+      braces[0] = `${getPrefix(constructor, tag) || 'DataView '}{`;
       // .buffer goes last, it's not a primitive like the others.
       keys.unshift('byteLength', 'byteOffset', 'buffer');
     } else if (isPromise(value)) {
-      braces[0] = `${getPrefix(constructor, tag)}{`;
+      braces[0] = `${getPrefix(constructor, tag) || 'Promise '}{`;
       formatter = formatPromise;
     } else if (isWeakSet(value)) {
-      braces[0] = `${getPrefix(constructor, tag)}{`;
+      braces[0] = `${getPrefix(constructor, tag) || 'WeakSet '}{`;
       if (ctx.showHidden) {
         formatter = formatWeakSet;
       } else {
         extra = '<items unknown>';
       }
     } else if (isWeakMap(value)) {
-      braces[0] = `${getPrefix(constructor, tag)}{`;
+      braces[0] = `${getPrefix(constructor, tag) || 'WeakMap '}{`;
```

Comment:
Would it make more sense to make the `|| 'thing '` be a param of `getPrefix` to add the trailing space then?